src/client/testing/common/xUnitParser.ts

```diff
@@ -49,10 +51,27 @@ export class XUnitParser implements IXUnitParser {
     constructor(@inject(IFileSystem) private readonly fs: IFileSystem) {}
 
     // Update "tests" with the results parsed from the given file.
-    public async updateResultsFromXmlLogFile(tests: Tests, outputXmlFile: string) {
-        const data = await this.fs.readFile(outputXmlFile);
+    public async updateResultsFromXmlLogFile(tests: Tests, outputXmlFile: string, timeout = 0) {
+        var data = '';
+        var parserResult = undefined;
+        const timer = new StopWatch();
+        // Poll for valid xml
+        do {
+            data = await this.fs.readFile(outputXmlFile);
```

Comment:
If the file might not be written yet, there's also the chance that it doesn't even exist yet.  So you may need to guard against that:

```suggestion
            try {
                data = await this.fs.readFile(outputXmlFile);
            } catch (err) {
                if (error.code === 'ENOENT') {
                    // The file hasn't been created yet.
                } else {
                    throw // re-throw
                }
            }
```